### PR TITLE
feat: passing dropzone icon as props

### PR DESCRIPTION
**Pull request by pefi1011**
Hi Yuvaleros,
what do you think of allowing to pass custom icon to the dropzone area? I did not test if it works because I want to see if you are open for PRs at all. 
This PR could be enriched with further improvements such as passing custom styles for the current CloudUploadIcon, updating docs, etc

### CODE REVIEW

**Comment by Yuvaleros** [comment]
Hi @pefi1011 , thank you for your contribution. Yes I'm open to PR's. If you could kindly add to the README file the relevant information on how to use the feature I'll approve the PR.

**Comment by max-carroll** [comment]
I think you should use React.createElement like this so that you can pass the classes in

image

then you can default it to the CloudUploadIcon like this

image

I have this working locally. Nice Idea to make the icon optional though

**Comment by max-carroll** [comment]
However perhaps its no big deal that you cant pass the classes in, it just means the developer can decide to set it to whatever they like

**Comment by panz3r** [comment]
Hi @max-carroll ,

Actually with the changes implemented by this PR you have the best customisability for the icon since you can pass in any `react` component (not just icons 😬 ), like this
```jsx
<DropzoneArea 
   icon={<AnyComponent className="sample-icon" ... />}
   onChange={this.handleChange.bind(this)}
/>
```

@pefi1011 recently there's been some update to the package code, could you please update this PR to resolve the conflicts?

**Comment by max-carroll** [comment]
Yes I agree actually good idea

**Comment by max-carroll** [comment]
Hi I know this isnt the best forum for this, but Im trying to help out on this project Im wondering what the best setup is for dev, I ran npm install and npm start in both the root directory and the example directory then I edit both the example and root source to see if it works. But sometimes I get funny errors and Im wondering if this is the best workflow setup. If someone just helps me get organized Ill do a bit of work on this 😀

**Comment by panz3r** [comment]
Hi @max-carroll ,

> Hi I know this isnt the best forum for this, but Im trying to help out on this project Im wondering what the best setup is for dev, I ran npm install and npm start in both the root directory and the example directory then I edit both the example and root source to see if it works. But sometimes I get funny errors and Im wondering if this is the best workflow setup. If someone just helps me get organized Ill do a bit of work on this 😀

Opened an issue to discuss/troubleshoot this, see #131 .

File: src/DropzoneArea.js [hunk 1]
```diff
@@ -202,7 +205,7 @@ class DropzoneArea extends Component {
                         <p className={classNames(classes.dropzoneParagraph, this.props.dropzoneParagraphClass)}>
                             {this.state.dropzoneText}
                         </p>
-                        <CloudUploadIcon className={classes.uploadIconSize} />
+                        {showIcon && dropzopneIcon}
```
Comment (max-carroll) [inline src/DropzoneArea.js:208]: typo with dropzoneIcon

File: src/DropzoneArea.js [hunk 2]
```diff
@@ -304,4 +308,8 @@ DropzoneArea.propTypes = {
     onDelete: PropTypes.func
 };
 
+DropzoneArea.defaultProps = {
```
Comment (max-carroll) [inline src/DropzoneArea.js:311]: Ahh, I think this is a little code is a little bit behind the current code now, requires the latest changes from master

**Comment by panz3r** [comment]
Superseded by #227

**Pull request closed without merging**